### ISSUE TITLE
Labels/Goto not working in Interactive (crashes VS with InvalidOperationException)

### ISSUE THREAD

**Issue by KevinH-MS** [dotnet/roslyn#3712]
1.  Open C# Interactive and paste the following:
```
C: \ a \b
```

**RESULTS:**
```
System.InvalidOperationException : This program location is thought to be unreachable.
	Binder\BuckStopsHereBinder.cs(208,0): at Microsoft.CodeAnalysis.CSharp.BuckStopsHereBinder.get_Labels()
	Binder\Binder_Statements.cs(3251,0): at Microsoft.CodeAnalysis.CSharp.Binder.get_Labels()
	Binder\Binder_Statements.cs(3251,0): at Microsoft.CodeAnalysis.CSharp.Binder.get_Labels()
	Binder\Binder_Statements.cs(3251,0): at Microsoft.CodeAnalysis.CSharp.Binder.get_Labels()
	Binder\Binder_Statements.cs(3251,0): at Microsoft.CodeAnalysis.CSharp.Binder.get_Labels()
	Compilation\MemberSemanticModel.cs(494,0): at Microsoft.CodeAnalysis.CSharp.MemberSemanticModel.GetDeclaredSymbol(LabeledStatementSyntax declarationSyntax, CancellationToken cancellationToken)
	Compilation\SyntaxTreeSemanticModel.cs(1598,0): at Microsoft.CodeAnalysis.CSharp.SyntaxTreeSemanticModel.GetDeclaredSymbol(LabeledStatementSyntax declarationSyntax, CancellationToken cancellationToken)
```

**Comment by KevinH-MS** [comment]
Note:  Per obsolete, internal tracking item (DEVDIV 522569), labels should be scoped to a file (SyntaxTree).  You should not be able to goto across script files (#include) and you should not be able to goto across submissions.

**Comment by codespare** [comment]
I see a related behaviour with csi.exe:
``` C#
   using static System.Console;
   if (true) {
      goto End;
   }
   End:
      WriteLine("This is the end.");
```
gives
> error CS0159: No such label 'End' within the scope of the goto statement

while
``` C#
   using static System.Console;
   {
      if (true) {
         goto End;
      }
      End:
         WriteLine("This is the end.");
   }
```
outputs the printed message.

**Comment by cston** [comment]
Fixed in #6076
